elf/elf.go

```diff
@@ -813,6 +834,58 @@ func perfEventPoll(fds []C.int) error {
 	return nil
 }
 
+// UpdateElement stores value in key in the map stored in mp.
+// The flags can have the following values (if you include "uapi/linux/bpf.h"):
+// C.BPF_ANY to create new element or update existing;
+// C.BPF_NOEXIST to create new element if it didn't exist;
+// C.BPF_EXIST to update existing element.
+func (b *Module) UpdateElement(mp *Map, key, value unsafe.Pointer, flags uint64) error {
+	uba := C.union_bpf_attr{}
+	C.create_bpf_update_elem(
+		C.int(mp.m.fd),
+		key,
+		value,
+		C.ulonglong(flags),
+		unsafe.Pointer(&uba),
+	)
+	ret, _, err := syscall.Syscall(
+		C.__NR_bpf,
+		C.BPF_MAP_UPDATE_ELEM,
+		uintptr(unsafe.Pointer(&uba)),
+		unsafe.Sizeof(uba),
+	)
+
+	if ret != 0 || err != 0 {
+		return fmt.Errorf("Unable to update element: %s", err)
```

Comment:
s/Unable/unable

https://github.com/golang/go/wiki/CodeReviewComments#error-strings

ditto in the other function